### PR TITLE
Add `wait` option to remote backends

### LINKED ISSUE #596: Wait option for Backend.run( ) function
Hello ! In the Cloud SDK, there is an option wait in sdk.create_batch. Setting wait = True keeps your code running until the results are available on the Cloud platform (i.e. batch is marked as Done). It is very useful to build a continuous communication loop between a local computer and the distant Emulator/QPU. It is possible to make the code artificially wait but since the option is already in the Cloud SDK and that the Backend objects are wrappers around that, I feel like it would be useful and easy to keep that argument. Thank you a lot !

### PR DESCRIPTION
**Pull request by HGSilveri**
Fixes #596 .

### CODE REVIEW

File: pulser-core/pulser/backend/qpu.py [hunk 1]
```diff
@@ -34,6 +36,7 @@ def run(self, job_params: list[JobParams] | None = None) -> RemoteResults:
                 is parametrized, the values for all the variables necessary
                 to build the sequence must be given in it's own mapping, for
                 each job, under the 'variables' field.
+            wait: Whether to wait until the results become available.
```
Comment (madagra) [inline pulser-core/pulser/backend/qpu.py:39]: Maybe add a bit more information: "Whether wait until the results of the jobs become available. If set to False, the call is non-blocking and the results must be checked using the `results()` property."

**Review by madagra — APPROVED** [review]
Looks good to me, just a minor comment.

File: pulser-core/pulser/backend/qpu.py (hunk 1 above)
Comment (a-corni) [reply pulser-core/pulser/backend/qpu.py:39]: Discussing with @LucasGitQ, we propose "Whether wait until the results of the jobs become available. If set to False, the call is non-blocking and the obtained results must be checked using their `status` property."

Comment (HGSilveri) [reply pulser-core/pulser/backend/qpu.py:39]: Thanks for the suggestions!

**Review by a-corni — APPROVED** [review]

**Pull request merged**